src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -402,6 +419,10 @@ private void OnBatchScopeDisposed()
                             var metadataReference = _workspace.CurrentSolution.GetProject(Id).MetadataReferences.Cast<PortableExecutableReference>()
                                                                                     .Single(m => m.FilePath == metadataReferenceRemovedInBatch.path && m.Properties == metadataReferenceRemovedInBatch.properties);
 
+                            _fileReferenceChangeContext.StopWatchingFile(_metadataReferenceFileWatchingTokens[metadataReference]);
```

Comment:
these 3 line seems repeated multiple times

Reply:
Going to leave it; to me the mental burden of a helper method with a single line is not worth it vs. being able to easily read the code in situ.